src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/PackageDeclarationCheck.java

```diff
@@ -72,12 +93,30 @@ public void finishTree(DetailAST ast) {
             if (ast != null) {
                 lineNumber = ast.getLineNo();
             }
-            log(lineNumber, MSG_KEY);
+            log(lineNumber, MSG_KEY_MISSING);
         }
     }
 
     @Override
     public void visitToken(DetailAST ast) {
         defined = true;
+
+        if (!ignoreMismatch) {
+            // Calculate the directory name, but stripping off the last
+            // part.
+            final String fileName = getFileContents().getFileName();
+            final int lastPos = fileName.lastIndexOf(File.separatorChar);
+            final String directoryName = fileName.substring(0, lastPos);
+
+            // Get the package name and format it like a directory name.
+            final DetailAST nameAST = ast.getLastChild().getPreviousSibling();
+            final FullIdent full = FullIdent.createFullIdent(nameAST);
+            final String expected = full.getText().replace('.', File.separatorChar);
```

Comment:
What does ```expected``` mean? Why ```full.getText().replace('.', File.separatorChar);``` is expected? 
Rename the variable. It should be as verbose as possible.